src/topology/algebra/ordered.lean

```diff
@@ -522,34 +521,54 @@ the left and to the right, either open or closed, i.e., members of `nhds_within
 characterized as the sets containing suitable intervals to the right or to the left of `a`.
 We give now these characterizations. -/
 
-/-- A set is a neighborhood of `a` within `(a, +∞)` if and only if it contains an interval `(a, u)`
-with `a < u`, provided `a` is not a top element. -/
-lemma mem_nhds_within_Ioi_iff_exists_Ioo_subset' {a u' : α} {s : set α} (hu' : a < u') :
-  s ∈ nhds_within a (Ioi a) ↔ ∃u, a < u ∧ Ioo a u ⊆ s :=
+-- NB: If you extend the list, append to the end please to avoid breaking the API
+/-- A few statements equivalent to `s ∈ nhds_within a (Ioi a)`. Most implications don't require
+`(a, +∞)` to be nonempty however if `(a, +∞)` is empty, then `nhds_within (a, +∞)=⊥`, and we
+don't care too much about this case. -/
+lemma tfae_mem_nhds_within_Ioi {a u' : α} (hu' : a < u') (s : set α) :
```

Comment:
Could you also give a version of this lemma in `no_top_order`, dropping the assumption `hu'`? I think this is the main use case.

Reply:
The only equivalence that makes sense without an explicit upper bound is in `mem_nhds_within_Ioi_iff_exists_Ioo_subset`. All other use `u'` in their types.